R/FormattingData.R

```diff
@@ -283,24 +331,60 @@ gen_background_pts <- function(nsdm_input, scale,
+                                                                 by = NULL)
+      }, error = function(e) {
+        Absences.XY <- unique(Absences.XY)
+        Absences.XY <- round(Absences.XY, digits = 4)
+        Absences.XY.final <- ecospat::ecospat.occ.desaggregation(Absences.XY,
+                                                                 min.dist = Min.Dist,
+                                                                 by = NULL)
+      })
+    }))
+    message(paste0(scale, " absence data (",SpeciesName,"): from ",
+                   nrow(Absences.XY), " to ", nrow(Absences.XY.final),
+                   " absence points after cleaning and thinning.\n"))
+
+    # Save filtered absences data for each species
+    if(save.output){
+    write.csv(Absences.XY.final, paste0("Results/", scale,
+                                        "/AbsencesXY/", SpeciesName, "_TrueAbsences.csv"))
+    }
+  }
+
   scale_summary <- data.frame(Values = c(SpeciesName,
                                          nrow(SpeciesData.XY),
                                          nrow(XY.final),
-                                         ifelse(is.null(background_scale),
+                                         ifelse(is.null(absences_scale),
+                                           ifelse(is.null(background_scale),
                                                 nPoints,
-                                                nrow(background_scale))))
+                                                nrow(background_scale)),
+                                           "NA"),
+                                        ifelse(is.null(absences_scale), "NA", nrow(Absences.XY)),
+                                        ifelse(is.null(absences_scale), "NA", nrow(Absences.XY.final))))
 
   rownames(scale_summary) <- c("Species name",
-                               paste("Original number of species occurrences at", lowcase_scale, "level"),
-                               paste("Final number of species occurrences at", lowcase_scale, "level"),
-                               paste("Number of background points at", lowcase_scale, "level"))
+                               paste0("Original number of species occurrences at ", lowcase_scale, " level"),
+                               paste0("Final number of species occurrences at ", lowcase_scale, " level"),
+                               paste0("Number of background points at ", lowcase_scale, " level"),
+                               paste0("Original number of true absence points at ", lowcase_scale, " level"),
+                               paste0("Final number of true absence points at ", lowcase_scale, " level"))
 
   return(list(
           SpeciesData.XY = XY.final,
-          Background.XY = Background.XY,
+          Background.XY = if(!is.null(Background.XY)) Background.XY else NULL,
+          Absences.XY = if(!is.null(absences_scale)) Absences.XY.final else NULL,
```

Comment:
Not sure why this conditional is needed. If it is not null so there is data in it, if it is null it will return a null regardless.